### PR TITLE
add hedron_compile_commands module to support compile_commands.json

### PR DESCRIPTION
**Pull request by chloro-pn**
chore(bazel) and refactor: 
* add hedron_compile_commands module to support compile_commands.json; 
* remove unused header files according to clangd's prompts；
* use async_simple::coro::CoroHandle

## What is changing
No incompatible changes

## Example
`bazel run @hedron_compile_commands//:refresh_all --  --action_env=CXX=clang++ --action_env=CC=clang --define ASYNC_SIMPLE_DISABLE_AIO=true`

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -6,3 +6,12 @@ bazel_dep(name = "bazel_skylib", version = "1.6.1")
...
 bazel_dep(name = "boringssl", version = "0.0.0-20240530-2db0eb3", dev_dependency = True)
+
+# Hedron's Compile Commands Extractor for Bazel
+# https://github.com/hedronvision/bazel-compile-commands-extractor
+bazel_dep(name = "hedron_compile_commands", dev_dependency = True)
+git_override(
+    module_name = "hedron_compile_commands",
+    remote = "https://github.com/hedronvision/bazel-compile-commands-extractor.git",
+    commit = "1e08f8e0507b6b6b1f4416a9a22cf5c28beaba93",
+)
```
Comment (4kangjc) [inline MODULE.bazel:17]: add new line

File: MODULE.bazel [hunk 2]
```diff
@@ -6,3 +6,12 @@ bazel_dep(name = "bazel_skylib", version = "1.6.1")
 bazel_dep(name = "googletest", version = "1.14.0.bcr.1", repo_name = "com_google_googletest", dev_dependency = True)
 bazel_dep(name = "google_benchmark", version = "1.8.4", repo_name = "com_google_benchmark", dev_dependency = True)
 bazel_dep(name = "boringssl", version = "0.0.0-20240530-2db0eb3", dev_dependency = True)
+
+# Hedron's Compile Commands Extractor for Bazel
+# https://github.com/hedronvision/bazel-compile-commands-extractor
```
Comment (4kangjc) [inline MODULE.bazel:11]: Write the usage instructions in README

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM

**Comment by ChuanqiXu9** [comment]
I'd like to land this. If there are following update or requests, we can make it in following commits.

**Pull request merged**